include/librdkafka/rdkafka.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * librdkafka - Apache Kafka C library
+ *
+ * Copyright (c) 2012-2022 Magnus Edenhill
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef RDKAFKA_RDKAFKA_H
+#define RDKAFKA_RDKAFKA_H
+
+#include <rdkafka.h>
```

Comment:
IIUC, the intent of this line is `#include "../../src/rdkafka.h"`; shouldn't it be spelled that way? Spelling it with angle brackets makes it seem like it would prefer `/usr/local/include/rdkafka.h`, which I think is not the intended connotation.

From first principles, I would have expected the _compatibility_ header in `./src/rdkafka.h` to be the one-liner `#include <librdkafka/rdkafka.h>`, and then `./include/librdkafka/rdkafka.h` would contain the actual code moving forward. But the commit messages/discussion sound like you considered that option and already rejected it? (I don't quite understand why, though.)

FWIW, for the non-compatibility headers that are actually going to be installed in `/usr/local/include` (which I would _hope_ means everything from `./include/` and nothing from `./src/`, even though that's not what this PR currently does), I strongly agree with the direction of this PR to use `#include <librdkafka/...>` consistently throughout. I just think the compatibility headers (which I would _hope_ means only things in `./src/`) should continue to use `#include "..."`.

Reply:
Actually, `#include "../../src/rdkafka.h"` did not come across my mind.

The motivation of having the compatibility header at `include/librdkafka/` was to maintain compatibility with projects which add `src/` and `src-cpp/` to their include dirs, but not `include/` with the actual header files. Adding `include/` to the include dirs would have been necessary, as the compatibility headers would have had `#include <librdkafka/rdkafka.h>` as their content.

I'm trying to rebuild this PR with `#include "../include/librdkafka/rdkafka.h"` later today.

Reply:
I see you've flipped the headers around per my suggestion (compatibility headers in `src/`, "real" headers in `include/librdkafka/`). Just for the record, yes, I now think you're doing it right. I don't _think_ I'm going to come back in a week and say "no no, I meant to suggest something radically different from that." :)  (And I hope this direction will be acceptable to @edenhill and also to whatever's going on with the VS project include directories. ;))